tests/app/dao/test_notification_dao.py

```diff
@@ -729,6 +729,67 @@ def test_updating_notification_updates_notification_history(sample_notification)
     assert hist.status == 'sending'
 
 
+def test_should_delete_notification_and_notification_history_for_id(notify_db, notify_db_session, sample_template):
+    data = _notification_json(sample_template)
+    notification = Notification(**data)
+
+    dao_create_notification(notification)
+
+    assert Notification.query.count() == 1
+    assert NotificationHistory.query.count() == 1
+
+    dao_delete_notifications_and_history_by_id(notification.id)
+
+    assert Notification.query.count() == 0
+    assert NotificationHistory.query.count() == 0
+
+
+def test_should_delete_only_notification_and_notification_history_with_id(notify_db, notify_db_session,
+                                                                          sample_template):
+    id_1 = uuid.uuid4()
+    id_2 = uuid.uuid4()
+    data_1 = _notification_json(sample_template, id=id_1)
+    data_2 = _notification_json(sample_template, id=id_2)
+
+    notification_1 = Notification(**data_1)
+    notification_2 = Notification(**data_2)
+
+    dao_create_notification(notification_1)
+    dao_create_notification(notification_2)
+
+    assert Notification.query.count() == 2
+    assert NotificationHistory.query.count() == 2
+
+    dao_delete_notifications_and_history_by_id(notification_1.id)
+
+    assert Notification.query.count() == 1
+    assert NotificationHistory.query.count() == 1
+    assert Notification.query.first().id == notification_2.id
+    assert NotificationHistory.query.first().id == notification_2.id
```

Comment:
if you change these lines from first to `.one()` then you don't need the count lines above, since .one will raise if there's not exactly one row